### ISSUE TITLE
TAQ Python data

### ISSUE THREAD

**Issue by juanhenao21** [juanhenao21/econophysics#3]
Implement a function in TAQ algorithm to obtain the TAQ data in Python format

**Comment by juanhenao21** [comment]
The following is the first implementation (V1) comment to load TAQ data and convert it to Python format.

The TAQ data is stored in a compressed way, and it is needed a `C++` to decompress the data. For my analysis, I only decompressed the AAPL and MSFT tickers in the `CSV` format to work with them in Python.

The TAQ data have two files, one with the trades information and another with the quoutes information. Each quotes file is really big (> 5 GB) as it cointains the information of a full year and the trades files are around ~ 1 GB, in our case we analyze the year 2008.

I try to load the data in different ways to avoid the memory of the PC overloads with this files, and I found a module in `Python` called `Dask` that is a flexible library for parallel computing in Python.

However, the results were not consistent, so I had to load the data and immediatly convert the interest columns in the dask dataframe to pandas to avoid the errors I was having. I would be interesting to analyze where was the exact point where the results just broke up, to obtain the best performance of dask (future task (?)).

Then I just save the columns of the pandas dataframe with some filters in numpy arrays to save the relevant information I would use in further analysis.
The final part of the code return a list with the days in each day that have data. Normally, on weekends the market is close and strangely there are some days in the week that there is no data.

**Comment by juanhenao21** [comment]
The TAQ data is stored in a compressed way, and it is needed a `C++` to decompress the data. For my analysis I only decompressed the AAPL and MSFT tickers in `CSV` format to work with them in Python.

The TAQ data have two files, one with the trades information and another with the quoutes information. Each quotes file is really big (> 5 GB) as it cointains the information of a full year and the trades files are around ~ 1 GB, in our case we analyze the year 2008.

I try to load the data in different ways to avoid the memory of the PC overloads with this files, and I found a module in `Python` called `Dask` that is a flexible library for parallel computing in Python.

However, the results were not consistent. In the second version I tried to improve the time to load and analyze the data. 

The following characteristics from V1 were making the implementation really slow

1. In the function `taq_data_extract` I was loading the full data every month, again and again, instead of load it just once.

2. I was computing the dask dataframe to convert it in pandas dataframe every month, but I think it is better to do it every day.

3. The algorithm is loading the data from the days that do not have data, and only when the day is used appear the error. It takes almost the same time it takes to a data day to be saved.

With this in mind I made an implementation (V2) where the data is loaded with dask but is computed every day. The test for one single day shows the following

* V1 = 3min 6s to load the data and compute the result to pandas. Then 52.7 s to select and save the required data. In total it used 3min 59s.
* V2 = 1min 27s to load the data. Then 1min 50s to select, compute and save the required data. In total it used 3min 17s.

As the result to select, compute and save the required data is almost 2 times slower than the V1, the results for the whole year will be slower. The results for one month are

* V1 = 2min 53s to load the data and compute the result to pandas. Then 45 s to select and save the required data. In total it used 25min 12s to analyze January.
* V2 = 1min 20s to load the data. Then 1min 40s to select, compute and save the required data. In total it used 53min 53s to analyze January.

In conclusion, the V2 is not the solution.

**Comment by juanhenao21** [comment]
The TAQ data is stored in a compressed way, and it is needed a `C++` to decompress the data. For my analysis I only decompressed the AAPL and MSFT tickers in `CSV` format to work with them in Python.

The TAQ data have two files, one with the trades information and another with the quotes information. Each quotes file is really big (> 5 GB) as it cointains the information of a full year and the trades files are around ~ 1 GB, in our case we analyze the year 2008.

I try to load the data in different ways to avoid the memory of the PC overloads with this files, and I found a module in `Python` called `Dask` that is a flexible library for parallel computing in Python.

However, the results were not consistent, so I had to load the data and immediatly convert the interest columns in the dask dataframe to pandas to avoid the errors I was having. 

In V1 I tought that the compile at the beginning was the big problem, but changing it in V2 did not show a better behavior, and instead, took almost the double of the time. With V2 I realized that I had the `taq_data_extract` in the wrong place, so I took it out from the inner `for loop` so the whole data only have to be loaded one time. There was also a problem with the load of dates that did not have data. I did not find a solution, but with the compute at the beginning, the identification of the `ValueError` could be faster.

The results for one day are

* V1 = 3min 6s to load the data and compute the result to pandas. Then 52.7 s to select and save the required data. In total it used 3min 59s.
* V2 = 1min 27s to load the data. Then 1min 50s to select, compute and save the required data. In total it used 3min 17s.
* V3 = 16min 1s to load the data and compute the result to pandas. Then 11m 45s to select and save the required data. In total it used 27min 50s.

The results for one month are

* V1 = 2min 53s to load the data and compute the result to pandas. Then 45 s to select and save the required data. In total it used 25min 12s to analyze January.
* V2 = 1min 20s to load the data. Then 1min 40s to select, compute and save the required data. In total it used 53min 53s to analyze January.
* V3 = it is a waste of time.

Making the compute for all the data require a lot of memory, and then the filter is really slow. Again, the first implementation has a better behavior.

In conclusionm the V3 is not the solution

**Comment by juanhenao21** [comment]
This version (V4) is the same as the V1, just some small adjusments as it was proved I can not find a better implementation.
I removed the `for loop` in the `data_to_array` function.

**Comment by juanhenao21** [comment]
I implement a parallelized version of V4, but it takes so much memory that the PC just freeze. The idea is at least use two cores. Hopefully it will work.

**Comment by juanhenao21** [comment]
It worked in my PC but not in the Uni PC. However it does not processing some days. I am reviewing the results with the original implementation to find if the `Try/Except` is failing.

**Comment by juanhenao21** [comment]
Update the TAQ data analysis file

The TAQ data analysis file has the implementations of the response and
the cross-correlator. I corrected the implementation of the trade signs
identification based on the implementation of S. Wang.

Some details in the docstring were added.

A revision to the self- and cross-correlator functions is needed
